source/crd.go

```diff
@@ -200,18 +198,23 @@ func (cs *crdSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 				}
 			}
 			if illegalTarget {
-				log.Warnf("Endpoint %s with DNSName %s has an illegal target. The subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com')", dnsEndpoint.Name, ep.DNSName)
-				continue
+				log.Warnf("Endpoint %s/%s with DNSName %s has an illegal target format.", dnsEndpoint.Namespace, dnsEndpoint.Name, ep.DNSName)
+				continue // Skip this specific endpoint
 			}
 
+			// Ensure labels are initialized
 			if ep.Labels == nil {
 				ep.Labels = endpoint.NewLabels()
 			}
 
+			// Add the valid endpoint to a temporary list for this CRD
 			crdEndpoints = append(crdEndpoints, ep)
 		}
 
+		// Add resource label to all valid endpoints from this CRD
```

Comment:
a desired solution is to add comment to `func (cs *crdSource) setResourceLabel` as it's more ideomatic way